bulbs/poll/models.py

```diff
@@ -0,0 +1,220 @@
+In the context of a single Poll:
+
+poll.models.Poll(pk=1)
+
+poll.models.Answer(pk=1) => answer_01
+poll.models.Answer(pk=2) => answer_02
+poll.models.Answer(pk=3) => answer_03
+
+And we delete Answer(pk=2)
+
+The next answer needs to be
+
+poll.models.Answer(pk=5) => answer_04
+
+So we cannot merely add up the answers of a poll, we
+have to keep track of how many answers we have already
+created.
+
+This is done by the field `last_answer_index` on the Poll model.
+
+It is incremented by 1 every time an Answer is saved.
+"""
+
+class SodaheadResponseError(APIException):
+    status_code = 503
+    default_detail = "Third-party poll provider temporarily unavailable."
+
+    def __init__(self, detail):
+        self.detail = detail
+
+class SodaheadResponseFailure(APIException):
+    status_code = 400
+    default_detail = "Error from third-party poll provider"
+
+    def __init__(self, detail):
+        self.detail = detail
+
+def Closed(): # noqa
+    end_date_params = {
+        "lte": timezone.now(),
+    }
+
+    return Range(end_date=end_date_params)
+
+class PollManager(ContentManager):
+    def search(self, **kwargs):
+        search_query = super(PollManager, self).search(**kwargs)
+
+        if "active" in kwargs:
+            search_query = search_query.filter(Published(before=timezone.now()))
+
+        if "closed" in kwargs:
+            search_query = search_query.filter(Closed())
+
+        return search_query
+
+class Poll(Content):
+    search_objects = PollManager()
+
+    question_text = models.TextField(blank=True, default="")
+    sodahead_id = models.CharField(max_length=20, blank=True, default="")
+    last_answer_index = models.IntegerField(default=0)
+    end_date = models.DateTimeField(null=True, default=None)
+    poll_image = ImageField(null=True, blank=True)
+
+    # This keeps Poll out of Content.search_objects
+    class Mapping(Content.Mapping):
+        poll_image = ElasticsearchImageField()
+        class Meta():
+            orphaned = True
+
+    def get_sodahead_data(self):
+        response = requests.get(SODAHEAD_POLL_ENDPOINT.format(self.sodahead_id))
+
+        if response.status_code != 200:
+            raise SodaheadResponseError(response.text)
+        else:
+            return response.json()
+
+    def sodahead_payload(self):
+        payload = {
+            'access_token': vault.read('sodahead/token')['value'],
+            'name': self.title,
+            'title': self.question_text,
+        }
+
+        if self.sodahead_id:
+            payload['id'] = self.sodahead_id
+
+        if self.published:
+            activation_date = self.published.astimezone(pytz.utc)
+            payload['activationDate'] = activation_date.strftime(SODAHEAD_DATE_FORMAT)
+
+        if self.end_date:
+            end_date = self.end_date.astimezone(pytz.utc)
+            payload['endDate'] = end_date.strftime(SODAHEAD_DATE_FORMAT)
+
+        for answer in self.answers.all():
+            if answer.answer_text == u'':
```

Comment:
Would `if not answer.answer_text` be appropriate here, or are you explicitly testing against an empty string? What if answer-text is `None`?